src/ucp/rma/flush.c

```diff
@@ -439,6 +450,12 @@ static void ucp_worker_flush_complete_one(ucp_request_t *req, ucs_status_t statu
 
     if (complete) {
         ucs_assert(status != UCS_INPROGRESS);
+
+        if (&req->flush_worker.next_ep->ep_list != &worker->all_eps) {
+            ep = ucp_ep_from_ext_gen(req->flush_worker.next_ep);
+            ucp_worker_flush_complete_ep(ep);
```

Comment:
move this into ucp_ep_flush_internal flow

Reply:
why it should be there? `ucp_worker_flush_complete_one()` can be called from `ucp_worker_flush_ep_flushed_cb()` as  well as from `ucp_ep_flush_internal()`

Reply:
/* cleanup ep iterator */
ucp_worker_flush_req_set_next_ep(&worker->all_eps)